spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/data/TestSparkAvroReader.java

```diff
@@ -20,45 +20,53 @@
 
 import static org.apache.iceberg.spark.data.TestHelpers.assertEqualsUnsafe;
 
-import java.io.File;
 import java.io.IOException;
 import java.util.List;
 import org.apache.avro.generic.GenericData.Record;
-import org.apache.iceberg.Files;
 import org.apache.iceberg.Schema;
 import org.apache.iceberg.avro.Avro;
 import org.apache.iceberg.avro.AvroIterable;
+import org.apache.iceberg.inmemory.InMemoryOutputFile;
 import org.apache.iceberg.io.FileAppender;
+import org.apache.iceberg.io.OutputFile;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 import org.apache.spark.sql.catalyst.InternalRow;
-import org.junit.Assert;
 
 public class TestSparkAvroReader extends AvroDataTest {
   @Override
   protected void writeAndValidate(Schema schema) throws IOException {
-    List<Record> expected = RandomData.generateList(schema, 100, 0L);
+    writeAndValidate(schema, schema);
+  }
+
+  @Override
+  protected void writeAndValidate(Schema writeSchema, Schema expectedSchema) throws IOException {
+    List<Record> expected = RandomData.generateList(writeSchema, 100, 0L);
 
-    File testFile = temp.newFile();
-    Assert.assertTrue("Delete should succeed", testFile.delete());
+    OutputFile outputFile = new InMemoryOutputFile();
```

Comment:
Spark 3.5 tests have been ported to JUnit 5, but 3.4 and 3.3 had not been. This required a few changes that I tried to keep to a minimum. This is one example, where the base class uses a Unit 4 `TemporaryFolder` that is not initialized for JUnit 5 parameterized tests.

Using `InMemoryOutputFile` here and in a couple other test suites avoided the need to port tests in #11811 to JUnit 5.

Reply:
Yeah I compared it side by side with the original, I think the changes made are reasonable